### PR TITLE
Simplify tk tooltip setup.

### PR DESCRIPTION
**Pull request by anntzer**
The whole module is private, so we can just clean it up directly.

Instead of creating and hiding a new tip window each time, we can just
create a single one and show/hide it as needed.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by QuLogic** [comment]
This causes some kind of deadlock if you mouse over buttons too quickly. I just moved left and right over the pan/zoom buttons and it reacts slower and slower until one of the tooltips stays visible and it stops changing to any other one. Sometimes it will unfreeze but going back over something else will get stuck again.

**Comment by anntzer** [comment]
I can't seem to repro this, either on Arch or on Fedora...  do you have a screencast?

**Comment by QuLogic** [comment]
This one is on `master`:

https://user-images.githubusercontent.com/302469/103111197-54c09a00-4618-11eb-98e6-8c85c207cf7b.mp4

Tooltips swap around fluidly and disappear as necessary.

This one is on this branch:

https://user-images.githubusercontent.com/302469/103111208-7883e000-4618-11eb-926c-ad03d5cedb8f.mp4

Whenever the tooltip shows up, it randomly causes freezes and mouse jitteriness.

**Comment by anntzer** [comment]
Ah, you may be using Wayland?  I can indeed repro with Wayland, not with X... I'll investigate more.

**Comment by QuLogic** [comment]
Yes I am; maybe this is an XWayland bug too?

**Comment by anntzer** [comment]
Indeed, this was easy enough to repro independently of matplotlib:
```python
import tkinter as tk

root = tk.Tk()

w = tk.Toplevel(root)
w.withdraw()

def enter(e):
    x = root.winfo_rootx()
    y = root.winfo_rooty()
    w.geometry(f"+{x+25}+{y+25}")  # overlapping windows => very slow
    # w.geometry(f"+{x+300}+{y+300}")  # non overlapping => no problem
    w.deiconify()

def leave(e):
    w.withdraw()

root.bind("<Enter>", enter)
root.bind("<Leave>", leave)

root.mainloop()
```

Do you have experience interacting with the wayland devs (I personally don't use wayland at all...)?  If so can you report this to them?  Otherwise I'll open an issue myself...

**Comment by anntzer** [comment]
OK, I went for a much less extensive cleanup, basically keeping the old logic.  The new commit message:

-----

Remove the separate createTooltip constructor, remove the unused x, y,
id attributes, and make showtip/hidetip directly usable as callbacks.

widget and text remain public attributes as one could imaging wanting to
update them (reparenting the tooltip or changing the text).

File: lib/matplotlib/backends/_backend_tk.py [hunk 1]
```diff
@@ -901,37 +901,23 @@ def set_history_buttons(self):
...
-        toolTip = ToolTip(widget)
-        def enter(event):
-            toolTip.showtip(text)
-        def leave(event):
-            toolTip.hidetip()
-        widget.bind('<Enter>', enter)
-        widget.bind('<Leave>', leave)
-
-    def __init__(self, widget):
+    def __init__(self, widget, text):
```
Comment (timhoffm) [inline lib/matplotlib/backends/_backend_tk.py:904]: Could have a short docstring explaining its use. Instantiating a class without binding it to a variable is an unusual construct. (I'm not even sure who holds the reference to the object, but I suppose, it's through the bound methods?)

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/backends/_backend_tk.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/backends/_backend_tk.py:904]: Yes, the callbacks maintain the ToolTip alive (this was already the case previously via the enter() and leave() callbacks).
I agree that instantiating the class without binding it is rather awkward, so I decided again to go back a closure-based approach like I originally tried (but keeping the repeated window create/deletions that existed in the previous code).

**Review by timhoffm — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Pull request merged**